Mastodon/Protocol/Provider/DataSourceFacade+Translate.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  DataSourceFacade+Translate.swift
+//  Mastodon
+//
+//  Created by Marcus Kida on 29.11.22.
+//
+
+import UIKit
+import CoreData
+import CoreDataStack
+import MastodonCore
+
+extension DataSourceFacade {
+    enum TranslationFailure: Error {
+        case emptyOrInvalidResponse
+    }
+    
+    public static func translateStatus(
+        provider: UIViewController & NeedsDependency & AuthContextProvider,
+        status: ManagedObjectRecord<Status>
+    ) async throws {
+        let selectionFeedbackGenerator = await UISelectionFeedbackGenerator()
+        await selectionFeedbackGenerator.selectionChanged()
+
+        guard
+            let status = status.object(in: provider.context.managedObjectContext)
+        else {
+            return
+        }
+        
+        func translate(status: Status) async throws -> String? {
```

Comment:
I haven't seen this "a function in a function" very often yet, but I can image that it makes sense from to time.

Totally regardless of this and more out of curiosity: How would you write unit tests for this? Wouldn't this be easier by extracting this into their own functions?

Reply:
I like to use such nested functions whenever the functions will only be used by their parents and not anywhere else, and yes they are not testable in isolation but only by invoking the parent function. I'll extract them and make them private to have it more consistent with the rest of the code.